crates/vm/levm/src/opcode_handlers/environment.rs

```diff
@@ -226,18 +226,30 @@ impl<'a> VM<'a> {
             return Ok(OpcodeResult::Continue { pc_increment: 1 });
         }
 
+        // Happiest fast path, copy without an intermediate buffer because there is no need to pad 0s and also size doesn't overflow.
+        let code_offset_end = code_offset.checked_add(size);
+        if code_offset_end.is_some() && code_offset_end < Some(current_call_frame.bytecode.len()) {
```

Comment:
nit: maybe I'm not getting something, but wouldn't it be a bit better to do something like:

```rust
if let Some(end) = code_offset.checked_add(size) && end < current_call_frame.bytecode.len() {
  // The same without the first line
}
```

It's a small difference but you avoid adding that two times, right?